src/routes/view-content/[guideId]/[bibleSection]/bible-menu/BookChapterSelectorPane.svelte

```diff
@@ -171,7 +183,27 @@
                 ? forceToInt(firstSelectedVerse.number)
                 : 0,
         };
-        setCurrentBibleSectionAndCurrentGuide(bibleSection, $currentPane?.guide ?? null);
+
+        if ($isPassageSearch) {
+            const languageId = get(currentLanguageInfo)?.id;
+            const allResources = (await fetchFromCacheOrApi(
+                ...searchResourcesEndpoint(
+                    languageId,
+                    '',
+                    [ParentResourceType.Images, ParentResourceType.Dictionary, ParentResourceType.Videos],
+                    bibleSection.bookCode,
+                    bibleSection.startChapter,
+                    bibleSection.startVerse,
+                    bibleSection.endChapter,
+                    bibleSection.endVerse
+                )
+            )) as ResourceContentInfo[];
```

Comment:
What would you think about moving this fetch to the LibraryMenu component? Having the pane responsible for the request here feels like the wrong place to me. If it was in the LibraryMenu you could do `setPassageSearchBibleSection` here and then in the LibraryMenu listen for the `$passageSearchBibleSection` change and trigger a fetch. Also would let us immediately close this pane and then use the existing loader spinner on the LibraryMenu.

Reply:
Would also let us move the `passageSearchResources` to the LibraryMenu so it doesn't need to live in the page context.